packages/kit/src/runtime/client/client.js

```diff
@@ -1165,6 +1163,21 @@ function get_navigation_intent(url, invalidating) {
 		// fall back to native navigation
 		return undefined;
 	}
+}
+
+/**
+ * Resolve the full info (which route, params, etc.) for a client-side navigation from the URL,
+ * taking the reroute hook into account. If this isn't a client-side-navigation (or the URL is undefined),
+ * returns undefined.
+ * @param {URL | undefined} url
+ * @param {boolean} invalidating
+ */
+function get_navigation_intent(url, invalidating) {
+	if (!url) return undefined;
+	if (is_external_url(url, base)) return;
+
+	const rerouted = get_navigation_url(url);
+	if (!rerouted) return undefined;
```

Comment:
don't need to write `return undefined` here. just `return` is equivalent and fewer bytes to send down the wire and parse
```suggestion
	if (!url) return;
	if (is_external_url(url, base)) return;

	const rerouted = get_navigation_url(url);
	if (!rerouted) return;
```